Source/WebCore/platform/SharedBuffer.cpp

```diff
@@ -669,11 +638,13 @@ void SharedBufferBuilder::initialize(Ref<FragmentedSharedBuffer>&& buffer)
     ASSERT(!m_buffer);
     // We do not want to take a reference to the SharedBuffer as all SharedBuffer should be immutable
     // once created.
-    if (buffer->hasOneRef() && !buffer->isContiguous()) {
+    if (buffer->hasOneRef() && buffer->isContiguous()) {
         m_buffer = WTFMove(buffer);
         return;
     }
     append(buffer);
+    if (m_buffer->segmentsCount() <= 1)
```

Comment:
I would tend to rewrite it by checking upfront isContiguous and either moving or cloning.
Otherwise, `m_buffer->m_contiguous = m_buffer->segmentsCount() <= 1`
Also, it seems `SharedBufferBuilder::append` is already setting m_contiguous.